ArcGISRuntimeSDKQt_CppSamples/Analysis/StatisticalQuery/StatisticalQuery.cpp

```diff
@@ -65,16 +65,19 @@ void StatisticalQuery::componentComplete()
   m_mapView->setMap(m_map);
 
   // connect to queryStatisticsCompleted
-  connect(m_featureTable, &ServiceFeatureTable::queryStatisticsCompleted, this, [this](QUuid, StatisticsQueryResult* result)
+  connect(m_featureTable, &ServiceFeatureTable::queryStatisticsCompleted, this, [this](QUuid, StatisticsQueryResult* rawResult)
   {
+    // Delete rawResult when we leave local scope.
+    QScopedPointer<StatisticsQueryResult> result(rawResult);
+
     // Iterate through the results
     QObject parent;
     QString resultText;
     StatisticRecordIterator iter = result->iterator();
     while (iter.hasNext())
     {
       StatisticRecord* record = iter.next(&parent);
-      for (auto key : record->statistics().keys())
+      for (const QString& key : record->statistics().keys())
```

Comment:
@anmacdonald this is another case where the iterators can be used just like with `StatisticalQueryGroupSort.cpp`

Reply:
Sorry missed this one! I'll do a search for ``keys()`` across the entire samples codebase.